blog/2025/issue-credentials-over-openid4vci.adoc

```diff
@@ -0,0 +1,376 @@
+- Universities issuing digital diplomas that employers can instantly verify for authenticity.
+- Companies issuing employee badges as verifiable credentials for office access or remote authentication.
+- Event organizers issuing verifiable tickets that can be validated offline.
+- Professional associations issuing membership credentials such as in our OAuth SIG example for verifying access to gated resources or conference venues.
+
+These examples illustrate the variety of situations where verifiable credentials remove the need for direct communication between verifier and issuer while preserving trust.
+
+== Configuring Keycloak for OpenID4VCI
+
+OpenID4VCI in Keycloak is offered via the feature flag `oid4vc-vci`, which consequently needs to be enabled at startup.
+
+[source,bash]
+----
+--features=oid4vc-vci
+----
+
+That said, we now assume that you have a running Keycloak instance with the `oid4vc-vci` feature flag enabled.
+
+Beyond enabling the feature flag, the following configuration steps need to be taken consistently, in accordance with your use case:
+
+- Configure OpenID4VCI at the realm level
+- Configure an issuable verifiable credential (as a dedicated client scope)
+- Enable and configure OpenID4VCI at the client level
+
+Because some configurations are not yet supported via the Keycloak Admin Console, we may often use the Admin REST API to communicate required configurations.
+
+=== Configuring OpenID4VCI at the realm level
+
+Let's assume you have already created a realm named `oauth-sig-realm` to represent the OAuth SIG group, and that a couple of users have also been created in this realm, each with a password assigned, to represent members of the group.
+
+In addition, any user who is intended to create credential offers must be granted the `credential-offer-create` role. This ensures that only authorized users can generate credential offers, maintaining the security of the issuance process.
+
+--
+++++
+<img src="${blogImages}/openid4vci/screenshot-realm-and-users.png" style="margin-bottom:2em" alt="Screenshot: Realm and Users">
+++++
+--
+
+The behavior of OpenID4VCI across the entire realm can be modulated via realm attributes.
+On the Admin Console, find the *OID4VCI Attributes* section under the *Realm Settings > Tokens* tab.
+
+--
+++++
+<img src="${blogImages}/openid4vci/screenshot-realm-config-oid4vci-attributes.png" style="margin-bottom:2em" alt="Screenshot: OID4VCI Attributes">
+++++
+--
+
+Sensible defaults apply out of the box, but for the sake of this demo, we would want, for instance, to increase the value of **Pre-Authorized Code Lifespan** to 3 minutes so that the issuance flow expires less quickly.
+To learn more about the **Nonce Lifetime** and other realm attributes for OpenID4VCI not accessible via the Admin Console, please refer to the main https://github.com/keycloak/keycloak/blob/main/docs/documentation/server_admin/topics/oid4vci/vc-issuer-configuration.adoc[Keycloak documentation].
```

Comment:
The nightly build of the archives and the container also comes with a nightly build of the documentation. So you can link there.

```suggestion
To learn more about the **Nonce Lifetime** and other realm attributes for OpenID4VCI not accessible via the Admin Console, please refer to the main https://www.keycloak.org/docs/nightly/server_admin/index.html#_oid4vci[Keycloak documentation].
```